pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -283,47 +283,16 @@ private boolean createDictionaryForColumn(ColumnIndexCreationInfo info, SegmentG
       return false;
     }
 
-    if (config.isOptimizeDictionary()) {
-
-      FieldIndexConfigs fieldIndexConfigs = config.getIndexConfigsByColName().get(column);
-      // Do not create dictionaries for json or text index columns as they are high-cardinality values almost always
-      if ((fieldIndexConfigs.getConfig(StandardIndexes.json()).isEnabled()
-          || fieldIndexConfigs.getConfig(StandardIndexes.text()).isEnabled())) {
-        return false;
-      }
-
-      // Do not create dictionary if index size with dictionary is going to be larger than index size without dictionary
-      // This is done to reduce the cost of dictionary for high cardinality columns
-      // Off by default and needs optimizeDictionary to be set to true
-      if (spec.isSingleValueField() && spec.getDataType().isFixedWidth()) {
-        return shouldCreateDictionaryWithinThreshold(info, config, spec);
-      }
-    }
-
-    if (config.isOptimizeDictionaryForMetrics() && !config.isOptimizeDictionary()) {
-      if (spec.isSingleValueField() && spec.getDataType().isFixedWidth() && spec.getFieldType() == FieldType.METRIC) {
-        return shouldCreateDictionaryWithinThreshold(info, config, spec);
-      }
-    }
-
-    return info.isCreateDictionary();
-  }
-
-  private boolean shouldCreateDictionaryWithinThreshold(ColumnIndexCreationInfo info,
-      SegmentGeneratorConfig config, FieldSpec spec) {
-    long dictionarySize = info.getDistinctValueCount() * (long) spec.getDataType().size();
-    long forwardIndexSize =
-        ((long) info.getTotalNumberOfEntries()
-            * PinotDataBitSet.getNumBitsPerValue(info.getDistinctValueCount() - 1) + Byte.SIZE - 1) / Byte.SIZE;
-
-    double indexWithDictSize = dictionarySize + forwardIndexSize;
-    double indexWithoutDictSize = info.getTotalNumberOfEntries() * spec.getDataType().size();
-
-    double indexSizeRatio = indexWithoutDictSize / indexWithDictSize;
-    if (indexSizeRatio <= config.getNoDictionarySizeRatioThreshold()) {
+    FieldIndexConfigs fieldIndexConfigs = config.getIndexConfigsByColName().get(column);
```

Comment:
nit: for easier readability, how about you set a variable defaulted to false and return that after the if condition?

e.g. 

```
boolean shouldCreateDictionary = false;
if (DictionaryIndexType.ignoreDictionaryOverride(config.isOptimizeDictionary(),
        config.isOptimizeDictionaryForMetrics(), config.getNoDictionarySizeRatioThreshold(), spec,
        fieldIndexConfigs, info.getDistinctValueCount(), info.getTotalNumberOfEntries())) {
    // Ignore overrides and pick from config
    shouldCreateDictionary = info.isCreateDictionary();
}
return shouldCreateDictionary;
```